hack/test-end-to-end-scenario.sh

```diff
@@ -0,0 +1,257 @@
+# when running as a restricted pod the registry will run with a pre-allocated
+# user in the neighborhood of 1000000+.  Look for a substring of the pre-allocated uid range
+oc exec -p ${frontend_pod} id | grep 10
+
+# Port forwarding
+echo "[INFO] Validating port-forward"
+oc port-forward -p ${frontend_pod} 10080:8080  &> "${LOG_DIR}/port-forward.log" &
+wait_for_url_timed "http://localhost:10080" "[INFO] Frontend says: " $((10*TIME_SEC))
+
+
+
+#echo "[INFO] Applying Docker application config"
+#oc create -n docker -f "${DOCKER_CONFIG_FILE}"
+#echo "[INFO] Invoking generic web hook to trigger new docker build using curl"
+#curl -k -X POST $API_SCHEME://$API_HOST:$API_PORT/osapi/v1beta3/namespaces/docker/buildconfigs/ruby-sample-build/webhooks/secret101/generic && sleep 3
+#os::build:wait_for_end "docker"
+#wait_for_app "docker"
+
+#echo "[INFO] Applying Custom application config"
+#oc create -n custom -f "${CUSTOM_CONFIG_FILE}"
+#echo "[INFO] Invoking generic web hook to trigger new custom build using curl"
+#curl -k -X POST $API_SCHEME://$API_HOST:$API_PORT/osapi/v1beta3/namespaces/custom/buildconfigs/ruby-sample-build/webhooks/secret101/generic && sleep 3
+#os::build:wait_for_end "custom"
+#wait_for_app "custom"
+
+echo "[INFO] Back to 'default' project with 'admin' user..."
+oc project ${CLUSTER_ADMIN_CONTEXT}
+
+# ensure the router is started
+# TODO: simplify when #4702 is fixed upstream
+wait_for_command '[[ "$(oc get endpoints router --output-version=v1beta3 -t "{{ if .subsets }}{{ len .subsets }}{{ else }}0{{ end }}" || echo "0")" != "0" ]]' $((5*TIME_MIN))
+
+echo "[INFO] Validating routed app response..."
+# use the docker bridge ip address until there is a good way to get the auto-selected address from master
+# this address is considered stable
+# used as a resolve IP to test routing
+CONTAINER_ACCESSIBLE_API_HOST="${CONTAINER_ACCESSIBLE_API_HOST:-172.17.42.1}"
+validate_response "-s -k --resolve www.example.com:443:${CONTAINER_ACCESSIBLE_API_HOST} https://www.example.com" "Hello from OpenShift" 0.2 50
+
+
+# Pod node selection
+echo "[INFO] Validating pod.spec.nodeSelector rejections"
+# Create a project that enforces an impossible to satisfy nodeSelector, and two pods, one of which has an explicit node name
+openshift admin new-project node-selector --description="This is an example project to test node selection prevents deployment" --admin="e2e-user" --node-selector="impossible-label=true"
+oc process -n node-selector -v NODE_NAME="${KUBELET_HOST}" -f test/node-selector/pods.json | oc create -n node-selector -f -
+# The pod without a node name should fail to schedule
+wait_for_command "oc get events -n node-selector | grep pod-without-node-name | grep failedScheduling"        $((20*TIME_SEC))
+# The pod with a node name should be rejected by the kubelet
+#wait_for_command "oc get events -n node-selector | grep pod-with-node-name    | grep NodeSelectorMismatching" $((200*TIME_SEC))
```

Comment:
possibly the `nodeName` being set on the pod doesn't match the actual name of the node?